lbz/resource.py

```diff
@@ -86,3 +91,86 @@ def pre_request_hook(self):
 
     def post_request_hook(self):
         pass
+
+
+class CORSResource(Resource):
+    _cors_headers = [
+        "Content-Type",
+        "X-Amz-Date",
+        "Authentication",
+        "Authorization",
+        "X-Api-Key",
+        "X-Amz-Security-Token",
+    ]
+
+    def __init__(self, event: dict, methods: List[str], origins: List[str] = None):
+        super().__init__(event)
+
+        self._allowed_origins = origins if origins else env.get("CORS_ORIGIN", "").split(",")
+        self._check_for_wildcard()
```

Comment:
Let's not mutate the state, rather set variables explicitly

How about:

```
    def __init__(self, event: dict, methods: List[str], origins: List[str] = None):
        super().__init__(event)

        allowed_origins = origins if origins else env.get("CORS_ORIGIN", "").split(",")
        self._allowed_origins = self._get_allowed_origin(allowed_origins)
        self._resp_headers = {
            ALLOW_ORIGIN_HEADER: self._allowed_origins[0],
            "Access-Control-Allow-Headers": ", ".join(self._cors_headers),
            "Access-Control-Allow-Methods": ", ".join([*methods, "OPTIONS"]),
        }

    def _get_allowed_origin(self, allowed_origins: List[str]) -> List[str]:
        actual_origin = self.request.headers.get("Origin")

        for allowed_origin in allowed_origins:
            if allowed_origin == actual_origin:
                return [actual_origin]
            elif "*" in allowed_origin:
                prefix, suffix = allowed_origin.split("*")
                if actual_origin.startswith(prefix) and actual_origin.endswith(suffix):
                    return [actual_origin]

        return allowed_origins
```

(it should be equivalent, not tested tho)